dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerImportService.java

```diff
@@ -102,32 +107,60 @@ public TrackerImportReport importTracker( TrackerImportParams params )
         {
             notifier.notify( params.getJobConfiguration(), "(" + params.getUsername() + ") Import:Start" );
         }
+        
+        try
+        {
 
-        List<TrackerBundle> trackerBundles = preheatBundle( params, importReport );
+            List<TrackerBundle> trackerBundles = preheatBundle( params, importReport );
 
-        trackerBundles = preProcessBundle( trackerBundles, importReport );
+            trackerBundles = preProcessBundle( trackerBundles, importReport );
 
-        TrackerValidationReport validationReport = validateBundle( params, importReport, trackerBundles );
+            TrackerValidationReport validationReport = validateBundle( params, importReport, trackerBundles );
 
-        if ( validationReport.hasErrors() && params.getAtomicMode() == AtomicMode.ALL )
-        {
-            importReport.setStatus( TrackerStatus.ERROR );
-        }
-        else
-        {
-            if ( TrackerImportStrategy.DELETE == params.getImportStrategy() )
+            if ( validationReport.hasErrors() && params.getAtomicMode() == AtomicMode.ALL )
             {
-                deleteBundle( params, importReport, trackerBundles );
+                importReport.setStatus( TrackerStatus.ERROR );
             }
             else
             {
-                commitBundle( params, importReport, trackerBundles );
+                if ( TrackerImportStrategy.DELETE == params.getImportStrategy() )
+                {
+                    deleteBundle( params, importReport, trackerBundles );
+                }
+                else
+                {
+                    commitBundle( params, importReport, trackerBundles );
+                }
             }
+
+            importReport.getTimings().setTotalImport( requestTimer.toString() );
+
+            TrackerBundleReportModeUtils.filter( importReport, params.getReportMode() );
+
         }
+        catch ( Exception e )
+        {
+            log.error( "Exception thrown during import.",e );
+            
+            TrackerValidationReport tvr = importReport.getTrackerValidationReport();
+            
+            if ( tvr == null )
+            {
+                tvr = new TrackerValidationReport();
+            }
+            
+            tvr.getErrorReports().add(new TrackerErrorReport( null, "Exception:"+e.getMessage(), TrackerErrorCode.E9999, 0, null, null, null, null ));
```

Comment:
There is a static method to create new errors ValidationErrorReporter.newReport(errorCode) and then you can add an arg.
You can find a lot of examples in the validation hooks.
Not a big change, but we avoid to have all this null in the constructor

Reply:
Good point! Thanks